### ISSUE TITLE
Text and Heading components don't seem to support theme variants

### ISSUE THREAD

**Issue by mitchmalone** [rebassjs/rebass#799]
I am trying to declare some _variants_ in my theme in order to try and standardize certain heading sizes, but as I apply the variants to `<Heading>` and `<Text>` components they don't render.

**Theme.js**
```
export default {
  breakpoints: [],
  fontSizes: [12, 14, 16, 20, 24, 32, 40, 42, 48, 52, 64],
  colors: {},
  space: [],
  fonts: {
    body: "Avenir",
    heading: "Avenir",
  },
  fontWeights: {
    body: 300,
    heading: 300,
    bold: 500,
  },
  lineHeights: {
    body: 1.38,
    heading: 1.33,
  },
  shadows: {},
  variants: {
    headingone: {
      bg: "red",
      fontSizes: [7, 8],
      fontWeight: 500,
      color: "#0000ff",
    }
  },
  text: {},
  buttons: {}
  },
};
```

**This variant works properly**
`<Box variant="headingone">This variant works properly</Box>`

**This variant does not work at all**
`<Heading variant="headingone">This variant does not work at all</Heading>`

**Comment by mitchmalone** [comment]
After reading through the docs it appears as though it's because the `<Heading>` component explicitly declares a variant of **'heading'**. Is it possible or in the roadmap to some how allow multiple variants on a single component to allow them to combine and merge?

**Comment by jxnblk** [comment]
By default Text and Heading use `theme.text` for variants (see the table here: https://rebassjs.org/variants), but you can specify any arbitrary variant by specifying the full path, e.g. `variant='variants.headingone'` -- multiple variants is something I'm looking into, but not currently supported

**Comment by focux** [comment]
Actually, variants for Heading aren't working either. I see that there was a bug related to that and it was fixed by this PR #754 but still hasn't been published.

**Comment by talentlessguy** [comment]
It is fixed in `4.0.7`, everything works! I think it is time to close the issue, @mitchmalone.

example:

```js
// theme.js
text: {
  heading: {
  color: 'black'  
  }
}
```

```js
// pages/index.js
import React from 'react'
import { Heading } from 'rebass'
export default () => <Heading>Hello</Heading>
